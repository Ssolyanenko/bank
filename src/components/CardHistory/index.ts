export * from './History';
export * from './Pagination';
export * from './CardHistory';
