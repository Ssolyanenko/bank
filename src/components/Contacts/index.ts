export * from './Contacts';
export * from './ContactsFooter';
export * from './ContactsInformation';
