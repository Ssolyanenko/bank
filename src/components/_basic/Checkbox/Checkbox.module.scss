@import 'src/styles/variables.module.scss';

.checkboxWrapper {
  display: flex;
  align-items: center;

  .checkBox {
    color: $grayDark_7;
  }

  [data-testid='CheckBoxIcon'],
  [data-testid='RadioButtonCheckedIcon'] {
    color: $orange;
    border-color: $orange;
  }
}
