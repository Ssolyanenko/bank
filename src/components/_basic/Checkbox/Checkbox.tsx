import { ChangeEvent, FC, ReactElement } from 'react';
import { Checkbox as CheckboxMUI } from '@mui/material';

import classes from './Checkbox.module.scss';

interface Props {
  name: string;
  onChange(event: ChangeEvent<HTMLInputElement>, isChecked: boolean): void;
  value?: boolean;
  children?: ReactElement | string;
  classTitle?: string;
}

export const Checkbox: FC<Props> = ({ name, onChange, value = false, children = '', classTitle = '' }) => (
  <div className={`${classes.checkboxWrapper} ${classTitle}`}>
    <CheckboxMUI value={value} onChange={onChange} name={name} checked={!!value} />
    {children}
  </div>
);
