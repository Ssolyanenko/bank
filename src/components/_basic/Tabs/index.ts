export * from './Tabs';
export * from './TabTitle';
