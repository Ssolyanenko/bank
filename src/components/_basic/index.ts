export * from './PrimaryButton';
export * from './Button';
export * from './Input';
export * from './InputPassword';
export * from './ModalWrapper';
export * from './ModalConfirmation';
export * from './PhoneCodeTimer';
export * from './ModalBox';
export * from './CloseButton';
export * from './BackButton';
export * from './StepperForm';
export * from './StepperLabel';
export * from './ChatButton';
export * from './BanTimer';
export * from './CardTemplate';
export * from './SecondaryButton';
export * from './generic';
export * from './SelectStyled';
export * from './TextFieldStyled';
export * from './InputStyled';
export * from './CardTemplate';
export * from './Modal';
export * from './SecureWrapper';
export * from './InputRange';
export * from './DropDown';
export * from './RadioButtonsGroup';
export * from './FormError';
export * from './Checkbox';
export * from './Switcher';
export * from './EditButton';
export * from './CardBlock';
export * from './Tabs';
export * from './ModalSuccess';
export * from './DropDownWithCheckbox';
export * from './SelectFormikTemplate';
export * from './Loader';
export * from './DatePicker';
export * from './CustomRoutes';
