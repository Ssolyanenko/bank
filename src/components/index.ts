export * from './LandingPageComponents';
export * from './Forms';
export * from './ArrowsBlock';
export * from './RootModal';
export * from './Slider';
export * from './Map';
export * from './Contacts';
export * from './MainPageComponents';
export * from './ExchangeRates';
export * from './CardProducts';
export * from './CardHistory';
export * from './SetPinSuccessModal';
export * from './MyCardsApplications';
export * from './MyLoans';
export * from './MyCardsApplications';
export * from './InformationBlock';
export * from './StatusTag';
export * from './DropDownWithSearchBranches';
export * from './FormInputBox';
export * from './MyCards';
export * from './ActivateBlockCardModal';
export * from './UserAccount';
export * from './SetPinCodeFormModal';
export * from './CardDetails';
export * from './CardStatement';
export * from './CardStatementButtons';
export * from './Accordion';
export * from './ReceiptButtons';
export * from './TemporaryComponent/TemporaryComponent';
export * from './TransferModal';
export * from './ModalChoosePaymentCard';
export * from './CardInfo';
export * from './ShareReceiptModal/ShareReceiptModal';
export * from './SocialMedia';
export * from './ResetFormsConfirmModal';
export * from './TransactionLimit';
export * from './ChangeTransactionLimit';
export * from './ActivateBlockCard';
export * from './CardManage';
export * from './ChangePINCode';
export * from './ResponseModal';
export * from './TransactionLimit';
