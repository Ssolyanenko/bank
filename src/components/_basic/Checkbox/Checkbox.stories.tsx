import { ChangeEvent, useState } from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import { Checkbox } from './Checkbox';

export default {
  title: 'Basic/Checkbox',
  component: Checkbox,
  argTypes: {
    value: {
      defaultValue: false,
      type: 'boolean',
      description: 'Value of the Checkbox',
    },
    onChange: {
      action: 'ChangeValue',
      type: { name: 'function' },
      description: 'Function to change value of the checkbox',
    },
    name: {
      defaultValue: 'checkboxName',
      type: 'string',
      description: 'Name of the field',
    },
    children: {
      defaultValue: 'I am checkbox',
      type: 'string',
      description: 'The label text of checkbox',
    },
    classTitle: {
      defaultValue: '',
      type: 'string',
      description: 'Additional className',
    },
  },
  parameters: {
    docs: {
      description: {
        component: 'Checkbox component with the label text inside it',
      },
    },
  },
} as ComponentMeta<typeof Checkbox>;

const Template: ComponentStory<typeof Checkbox> = ({ name, classTitle, children, onChange, value }) => {
  const [isChecked, setIsChecked] = useState(value);

  const handleOnChange = (event: ChangeEvent<HTMLInputElement>, isChecked: boolean): void => {
    setIsChecked(isChecked);
    onChange(event, isChecked);
  };

  return (
    <Checkbox
      name={name}
      value={isChecked}
      onChange={(event: ChangeEvent<HTMLInputElement>): void => handleOnChange(event, event.target.checked)}
      classTitle={classTitle}
    >
      {children}
    </Checkbox>
  );
};

export const Default = Template.bind({});
