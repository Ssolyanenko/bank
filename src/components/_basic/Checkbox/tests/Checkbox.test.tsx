import 'jsdom-global/register';
import React, { ChangeEvent } from 'react';
import { mount, shallow } from 'enzyme';
import { Checkbox as CheckboxMUI } from '@mui/material';

import { Checkbox } from 'components/_basic';
import { Default } from 'components/_basic/Checkbox/Checkbox.stories';

describe('<Checkbox />', (): void => {
  test('should make a snapshot of component', (): void => {
    const props = {
      name: 'checkbox',
      onChange: jest.fn(),
    };
    const wrapper = shallow(<Checkbox {...props} />);
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });

  test('should change check', (): void => {
    const onCheckMock = jest.fn();
    const props = {
      name: 'checkbox',
      onChange: onCheckMock,
    };
    const wrapper = shallow(<Checkbox {...props} />);
    wrapper.find(CheckboxMUI).simulate('change', {});

    expect(onCheckMock).toHaveBeenCalled();
  });

  test('should render Checkbox from storybook', (): void => {
    const props = {
      name: 'checkbox',
      onChange: jest.fn(),
      value: true,
    };

    const wrapper = shallow(<Default {...props} />);

    expect(wrapper.find(Checkbox).length).toBe(1);
  });

  test('should handle storybook checkbox change', (): void => {
    let valueMock = false;
    const setValueMock = jest.fn((value) => {
      valueMock = value;
    });
    jest.spyOn(React, 'useState').mockImplementation(() => [valueMock, setValueMock]);

    const props = {
      name: 'checkbox',
      onChange: ({ target: { checked } }: ChangeEvent<HTMLInputElement>): void => setValueMock(checked),
      value: valueMock,
    };

    const wrapper = mount(<Default {...props} />);
    wrapper.find('input').simulate('change', { target: { checked: true } });

    expect(valueMock).toEqual(true);
  });
});
