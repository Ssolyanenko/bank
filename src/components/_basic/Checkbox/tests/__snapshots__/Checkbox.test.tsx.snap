// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should make a snapshot of component 1`] = `
<div
  className="checkboxWrapper "
>
  <ForwardRef(Checkbox)
    checked={false}
    name="checkbox"
    onChange={[MockFunction]}
    value={false}
  />
  
</div>
`;
